auth_proxy/proxy/flask.py

```diff
@@ -34,7 +52,27 @@ def __init__(self, url, orig):
     def request(self):
         """ @inherit
         """
-        url = self.url + '?' + parse.urlencode(self.orig.args)
+        args = dict(self.orig.args)
+
+        # Get the authorized scopes from the request, or open everything if
+        # this is an un-authorized request
+        try:
+            scopes = list(self.orig.oauth.scopes)
+        except AttributeError:
+            scopes = ['patient/*.read',]
+
+        # Wildcard scopes should be expanded
+        try:
+            wildcard = scopes.index('patient/*.read')
+            scopes[wildcard:wildcard] = self.ccds_scopes
+        except ValueError:
+            pass
+
+        # Update the query params
+        args['_security'] = ' '.join(scopes + ['public'] + args.get('_security', ''))
```

Comment:
Your _logic_ is not flawed — but I think the mental model of "proxy controls security" is much simpler and it avoids the opportunity for any kind of shenanigans if a tricky client tries to manipulate these values.